src/lib/storage/proposal/partition_creator.rb

```diff
@@ -87,9 +87,15 @@ def total_free_size
 
         # List of free spaces in the devicegraph
         #
+        # @param TODO: documentation
         # @return [FreeDiskSpacesList]
-        def free_spaces
-          candidate_disks.free_disk_spaces.with do |space|
+        def free_spaces(disk_name = nil)
```

Comment:
Does not look surprising to me. Without parameters it lookups all the devices. With an optional parameter you can restrict the scope of the search.

Any suggestion?

PS.- Yes, I got the reference :wink: But this case is sooo different... :smile: